report/src/main/java/com/hartwig/actin/report/interpretation/MolecularDriverEntryFactory.java

```diff
@@ -145,57 +135,36 @@ private Set<MolecularDriverEntry> fromDisruptions(@NotNull Set<Disruption> disru
 
             addActionability(entryBuilder, disruption);
 
-            entries.add(entryBuilder.build());
-        }
-
-        return entries;
+            return entryBuilder.build();
+        });
     }
 
     @NotNull
-    private Set<MolecularDriverEntry> fromFusions(@NotNull Set<Fusion> fusions) {
-        Set<MolecularDriverEntry> entries = Sets.newHashSet();
-
-        for (Fusion fusion : reportable(fusions)) {
+    private Stream<MolecularDriverEntry> fromFusions(@NotNull Set<Fusion> fusions) {
+        return fusions.stream().filter(Driver::isReportable).map(fusion -> {
             ImmutableMolecularDriverEntry.Builder entryBuilder = ImmutableMolecularDriverEntry.builder();
             entryBuilder.driverType(fusion.driverType().display());
             entryBuilder.driver(fusion.event() + ", exon " + fusion.fusedExonUp() + " - exon " + fusion.fusedExonDown());
             entryBuilder.driverLikelihood(fusion.driverLikelihood());
 
             addActionability(entryBuilder, fusion);
 
-            entries.add(entryBuilder.build());
-        }
-
-        return entries;
+            return entryBuilder.build();
+        });
     }
 
     @NotNull
-    private Set<MolecularDriverEntry> fromViruses(@NotNull Set<Virus> viruses) {
-        Set<MolecularDriverEntry> entries = Sets.newHashSet();
-
-        for (Virus virus : viruses) {
+    private Stream<MolecularDriverEntry> fromViruses(@NotNull Set<Virus> viruses) {
+        return viruses.stream().map(virus -> {
```

Comment:
`fromViruses` is the only case where we weren't filtering using `isReportable()`, so I didn't change that. Do we still want to use this filter for other drivers? Should it be applied to Viruses as well? Should we also filter with `isReportable()` on the summary page for consistency?